crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleProof.kt

```diff
@@ -31,6 +31,6 @@ interface MerkleProof {
      */
     fun verify(
         root: SecureHash,
-        digestProvider: MerkleTreeHashDigestProvider
+        digestProvider: MerkleTreeHashDigest
```

Comment:
@alexey-kadyrov-r3 in corda-runtime-os, this is a bit of an issue because it needs to call methods which don't exist in the marker interface. Is this where the proxying you mentioned needs to happen? The analogy for this was `DigitalSignatureVerificationServiceImpl`, but there we inject the proxy service, which doesn't seem possible here.

Reply:
Update: just had a quick chat with @mnesbit who explained that it's ok to just cast in this situation, since we can expect that it's safe to do so.